### DISCUSSION TITLE
About watch failure after ETCD restarts

### DISCUSSION THREAD

**Discussion by yuyu333333** [etcd-io/jetcd#1135] [Q&A]
After the ETCD restarts, the watch function fails， modify the values stored in ETCD, I cannot trigger the watch.

**Comment by lburgazzoli** [comment]
can you provide some more details ? a reproducer would be more than welcome

**Reply by yuyu333333** [reply]
When running ETCD normally, the watch function is effective. But when the ETCD server is restarted, the original watch becomes invalid. I operate as shown in the picture
[
![image](https://github.com/etcd-io/jetcd/assets/132212277/556bb14d-6bf6-4bb1-bb1f-59dc1bba1ffa)
](url)

**Comment by lburgazzoli** [comment]
can you create a reproducer in a form of an integration test ?

**Reply by yuyu333333** [reply]
When I am running the project and ETCD normally, the watch is effective; At this point, restart etcd (as shown in the previous screenshot) and the watch will fail. When the data in ETCD changes, the following image cannot be watched again

![image](https://github.com/etcd-io/jetcd/assets/132212277/d123be7b-f060-446b-9e76-ecef0d1f3701)

**Comment by lburgazzoli** [comment]
I think I got it but what I'm asking is some help in creating a test case so I can look into it quickly (I have very limited time)

**Comment by liangyuanpeng** [comment]
I have tested it and nothing wrong.

etcd version: 3.4.27 & 3.5.9
jetcd version: 0.7.5

Could you please help to confirm that? @yuyu333333 


test steps:

1. watch key /hello
2. put /hello world
3. got the watch result
4. restart etcd
5. put /hello world2
6. got the watch result

**Reply by liangyuanpeng** [reply]
ping @yuyu333333 

Same with https://github.com/etcd-io/jetcd/issues/1227
